modules/wallet/defrag.go

```diff
@@ -78,11 +81,18 @@ func (w *Wallet) createDefragTransaction() ([]types.Transaction, error) {
 	}
 
 	// Create the defrag transaction.
-	fee := defragFee()
 	refundAddr, err := w.nextPrimarySeedAddress(w.dbTx)
 	if err != nil {
 		return nil, err
 	}
+
+	// get the transaction fee
+	w.mu.Unlock()
+	minFee, _ := w.tpool.FeeEstimation()
+	sizeAvgOutput := uint64(250)
+	fee := minFee.Mul64(sizeAvgOutput * defragBatchSize)
+	w.mu.Lock()
```

Comment:
I worry about patterns like these. When you see a `lock; defer unlock;` at the top of a function, you generally make assumptions about things that you can modify, and about consistency. The fact that it actually releases the lock midway through the function can be a problem, especially if you write code that's expecting values in the wallet not to change.

Reply:
It's definitely a much safer pattern to grab the `minFee` before you lock the wallet at all.